tests/conftest.py

```diff
@@ -2517,12 +2519,18 @@ def awscli_pod_fixture(request, scope_name):
         pod: A pod running the AWS CLI
 
     """
+    project = f"namespace-{get_random_str()}"
+    ocp_obj = ocp.OCP(namespace=project)
+
+    def delete_project(namespace):
+        if "openshift" not in namespace:
+            ocp_obj.delete_project(project)
 
-    request.addfinalizer(awscli_pod_cleanup)
+    request.addfinalizer(lambda: delete_project(namespace=project))
+    request.addfinalizer(lambda: awscli_pod_cleanup(namespace=project))
 
-    log.info("Cleaning up any previous AWS CLI resources")
-    awscli_pod_cleanup()
-    return create_awscli_pod(scope_name)
+    ocp_obj.new_project(project)
```

Comment:
This has a side effect of changing the default project to the newly created one. I don't think we're running any oc commands without specifying the namespace/project in ocs-ci, but please change the default project back to `ENV_DATA["cluster_namespace"]` to be on the safe side.

Reply:
done

Reply:
Using `ocp.switch_to_default_rook_cluster_project` in the finalizer means that we'd only switch back to the default namespace after the tests that use this fixture are done, or at the end of the session when we're using this fixture with the session scope. 

Using this method right after the creation of the new project instead should guarantee we avoid this side effect:
```
    request.addfinalizer(lambda: delete_project(namespace=project))
    request.addfinalizer(lambda: awscli_pod_cleanup(namespace=project))

    ocp_obj.new_project(project)
    ocp.switch_to_default_rook_cluster_project()
```

Reply:
done